### ISSUE TITLE
[Feature request] Event when player equip or un-equip any armor

### ISSUE THREAD

**Issue by caoli5288** [PaperMC/Paper#928]
# Suggestion
Other mean's a event fire when player's armor contends has any change by any ways. :)

**Comment by 0-x-2-2** [comment]
You can already do this without needing a new event with the current events if you really want you can create the event in your plugin

**Comment by caoli5288** [comment]
@w-o-a-h listen about 5 events for 1 basement behavior？

**Comment by pkt77** [comment]
It is definitely more efficient to do this native. I have this in a private repo but it's only for monitoring, no cancel. Needed for custom enchants.

**Comment by MiniDigger** [comment]
@pkt77 if you have something like this already, do you care to submit it as a pull request so everybody can use it?

**Comment by pkt77** [comment]
@MiniDigger I have no clue how to work with patch-based repos, sorry. Would you be able to if I gave a pastebin?

**Comment by MiniDigger** [comment]
@pkt77 there are many ppl in papers irc channel who are happy to assist you with the process if you got some time to spare, else you can just drop a link to pastebin here and somebody can take a look at it.

**Comment by pkt77** [comment]
Here's my code if you guys want to make a patch out of it: https://pastebin.com/cZFJbxST

**Comment by meyden92** [comment]
That would save me a lot of lines and listeners in many projects. Would love to see that implemented/added. 

@w-o-a-h It's always good to have things implemented to make development easier/faster, and an armor equip event is kinda easier/fast than listening to tons of events c:

**Comment by BillyGalbreath** [comment]
I'd do it, but I still have PRs that havent been accepted and its made me lose interest in the project. ~~But thanks for the code. I'll be using it in my Forge fork ^_^~~ Nvm, Forge already has it :3

**Comment by zachbr** [comment]
Thanks.

I've made some minor housekeeping changes. Mostly just javadocs, nullability annotations, and I had to move it to our package. We don't add to spigot's package and I'm really pushing to get us out of bukkit's packages as well.

It's still basically all your work, I do not have time to get this implemented across all potential environmental factors so it's still not cancellable or called for dispensers or anything.
